### ISSUE TITLE
TSQL convert function incorrect conversion to sql string

### ISSUE THREAD

**Issue by MikeWallis42** [tobymao/sqlglot#3023]
I'm using the TSQL dialect and can confirm that it's an issue with the latest version 21.2.1

```python
from sqlglot import select
from sqlglot.dialects.tsql import TSQL

select_statement = select(
    "CONVERT(varchar, '2023-01-01 23:10:11.123', 121) AS current_datetime", dialect=TSQL
).from_("my_table")
sql = select_statement.sql(dialect=TSQL)
print(sql)
```

Output

```tsql
 SELECT CAST(FORMAT('2023-01-01 23:10:11.123', 'yyyy-MM-dd HH:mm:ss.S') AS VARCHAR(30)) AS current_datetime FROM my_table
```

Which isn't valid TSQL, apparently you can't pass it a varchar in the first position, you have to do something like this apparently with format.
Note .S should be .f repeated to however many digits of precision you need.

```tsql
DECLARE @d DATETIME = '2023-01-01 23:10:11.123'
SELECT CAST(FORMAT(@d, 'yyyy-MM-dd HH:mm:ss.fff') AS VARCHAR(30)) AS current_datetime
```

It would probably be much better that it not try and optimise the original convert statement which can take a varchar.

**Comment by georgesittas** [comment]
Should have a solution for this one tomorrow, will probably try to preserve `CONVERT` in T-SQL and do a best-effort transformation to transpile it elsewhere. Appreciate the report!

**Comment by MikeWallis42** [comment]
Thank you so much for jumping on this one, I can't wait to remove the hacky workaround I put in place 👍

**Comment by georgesittas** [comment]
@MikeWallis42 the fix for this has been deployed with v22.1.0
